src/averageAges.js

```diff
@@ -9,17 +9,25 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
+ *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter(person => {
+      const isMan = person.sex === 'm';
+      const fitCentury = Math.ceil(person.died / 100) === century;
+
+      return isMan && fitCentury;
+    })
+    : people.filter(person => person.sex === 'm');
```

Comment:
```suggestion
  const men = people.filter(person => century 
    ? person.sex === 'm' && Math.ceil(person.died / 100) === century;
    : person.sex === 'm'
  );
```

Reply:
I intentionally left these codes in this form. because it is important for me to remember the algorithm for solving these problems.